libminifi/include/core/repository/VolatileFlowFileRepository.h

```diff
@@ -55,7 +55,9 @@ class VolatileFlowFileRepository : public VolatileRepository<std::string> {
           std::shared_ptr<FlowFileRecord> eventRead = std::make_shared<FlowFileRecord>(shared_from_this(), content_repo_);
           if (eventRead->DeSerialize(reinterpret_cast<const uint8_t *>(purgeItem.data()), purgeItem.size())) {
             std::shared_ptr<minifi::ResourceClaim> newClaim = eventRead->getResourceClaim();
-            content_repo_->remove(newClaim);
+            if (newClaim != nullptr) {
+              content_repo_->removeIfOrphaned(newClaim);
```

Comment:
By not choosing the removeIfOrphaned path it
 - could delete a content claim when it still had references to it
 - bypassed the ContentRepository's internal bookkeeping, making it eat more and more RAM.